### ISSUE TITLE
Failed to find a kernelspec to use for ipykernel launch

### ISSUE THREAD

**Issue by Salvi07** [microsoft/vscode-jupyter#613]
## Environment data

-   VS Code version: 1.50.0-insider (user setup)
-   Extension version (available under the Extensions sidebar):2020.9.106814-dev
-   OS and version:  Windows_NT x64 10.0.18362
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.5 32-bit
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
 Jupyter 1.1.5, 

[**NOTE**: If you suspect that your issue is related to the Microsoft Python Language Server (`python.languageServer: 'Microsoft'`), please download our new language server [Pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) from the VS Code marketplace to see if that fixes your issue]

## Expected behaviour
Opening Jupyter files and be able to run code

## Actual behaviour
It shows.  Failed to find a kernelspec to use for ipykernel launch
![Screenshot (3)](https://user-images.githubusercontent.com/71162554/92986684-4cbc6e80-f4ef-11ea-8279-36e21340ff00.png)

## Steps to reproduce:

I open Jupyter in VSCode insider then this shows up "Failed to find a kernelspec to use for ipykernel launch"

## Logs

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by christoerasmus** [comment]
The issue appears to be back in version: v2020.12.414227025

![image](https://user-images.githubusercontent.com/41998773/107482548-81a91c00-6b88-11eb-89c3-443104dc667c.png)

**Comment by Th3OnlyN00b** [comment]
Can second this, appears to be back in v2020.12.414227025

**Comment by vikramjit-sidhu** [comment]
I concur, I am facing the same error message and issue.

**Comment by Gamezordd** [comment]
The following seems to have gotten rid of the error for me, occurred on a fresh install:

`
pip install ipython
pip install ipykernel
`

**Comment by Gamezordd** [comment]
The following seems to have gotten rid of the error for me, occurred on a fresh install:

`
pip install ipython  
`
`
pip install ipykernel
`

If you already have a kernel installed, upgrading it may work too. Try:

`pip install --upgrade ipython`
`pip install --upgrade ipykernel`

**Comment by Jiraiya854** [comment]
i've fixed the problem as well using this : 
pip install --upgrade ipython
pip install --upgrade ipykernel

**Comment by TogrulKazimov** [comment]
I am still facing this issue?

**Comment by SRKabir** [comment]
I was facing this issue consistently, and restarting ipykernel would often solve the problem. However, lately nothing would work (i.e. uninstalling and reinstalling python, ipython, ipykernel, upgrading ipython,ipykernel). I'm using VSCode insider and ssh'ed to remote machine.. any help please?

**Comment by DonJayamanne** [comment]
@TogrulKazimov
Please could you create a separate issue with the logs and detailed of your environment.

**Comment by doubleplusnice** [comment]
Issue seemingly persists. Found no way workaround so far

**Comment by DonJayamanne** [comment]
@doubleplusnice
Please could you create an issue with the details.
I'd like to get you unblocked

**Comment by suketubhavsar** [comment]
> The issue appears to be back in version: v2020.12.414227025
> 
> ![image](https://user-images.githubusercontent.com/41998773/107482548-81a91c00-6b88-11eb-89c3-443104dc667c.png)

Same error with VS Code, started VS Code as administrator and that fixed the issue. It seems I used "all users" setup while installing Anaconda could be the reason VSCode could not load python kernel.

**Comment by tjucaokun** [comment]
Same error with VSCode, i use the jupyter-extension in 2021.5.702919634,it cannot find the envs in my Anaconda @DonJayamanne

**Comment by AGilarde12** [comment]
@tjucaokun you can hopefully remedy this by changing where your VSCode is pointing in terms of the local host. 

First launch the Jupyter Notebook command prompt by searching "Jupyter" in your search. Then after it loads the info you need will be under "To access the notebook". Grab the http://localhost address and go to VSCode

Second hit this icon:
![image](https://user-images.githubusercontent.com/29666438/113953064-f85b4200-97e4-11eb-985b-eb63fb630afe.png)

And add the address you just copied as a connection endpoint. Then it'll ask you to refresh the session and you should be good.

I ran into this today at work but realized it was just pointing at the wrong notebook session. After that you should be all set.

**Comment by DonJayamanne** [comment]
@suketubhavsar, @AGilarde12,@tjucaokun
Please could you file a new issue with the details of the error and environment.
I'd like to get you unblocked. 

This issue had been closed for a while,  i.e. I want to help you get unblocked.

**Comment by chuan-325** [comment]
> The following seems to have gotten rid of the error for me, occurred on a fresh install:
> 
> `pip install ipython `
> `pip install ipykernel`
> 
> If you already have a kernel installed, upgrading it may work too. Try:
> 
> `pip install --upgrade ipython`
> `pip install --upgrade ipykernel`

Thanks! I fixed my same issue to this thread by your method.
It should be known by other users who are searching for solutions.
